tests/integration/cli/utils.py

```diff
@@ -0,0 +1,118 @@
+from pathlib import Path
+from typing import Callable, List
+
+import pytest
+
+projects_directory = Path(__file__).parent / "projects"
+project_names = [p.stem for p in projects_directory.iterdir() if p.is_dir()]
+
+
+def assert_failure(result, expected_output):
+    assert result.exit_code == 1
+    assert result.exception is not None
+    assert "ERROR" in result.output
+    assert expected_output in result.output
+
+
+class NodeId:
+    """
+    A class that extracts a callable test's name and module name.
+    """
+
+    def __init__(self, test_method: Callable):
+        self.module_full_name = test_method.__module__  # type: ignore
+        self.name = test_method.__name__
+
+    @property
+    def module_name(self) -> str:
+        return self.module_full_name.split(".")[-1]
+
+    @property
+    def node_id(self) -> str:
+        return f"{self.module_name}.{self.name}"
+
+
+class ProjectSkipper:
+    """
+    A class that contains stateful information about
+    which projects to skip tests for.
+    """
+
+    def __init__(self):
+        self.projects = {n: {} for n in project_names}
+
+    def __iter__(self):
+        return iter(self.projects)
+
+    def do_skip(self, project: str, module: str, test: str) -> bool:
+        """
+        Returns ``True`` if a test has been marked to be
+        skipped for the given project using the ``skip_project`` or
+        ``skip_project_except`` decorators.
+        """
+        result = test in self.projects[project].get(module, [])
+        return result
+
+    def _raise_if_not_exists(self, project_name: str, node_id: str):
```

Comment:
@fubuloubu Any project specified that does not exist will raise an error, that should be helpful when writing tests if you butcher the name of a project or something. 

This addresses that part of your feedback from another comment! :)